### PR TITLE
Upgrading to 2.12.0 and using factoryFor over _lookupFactory

### LINKED ISSUE #67: Throwing `_lookupFactory` deprecation warning in Ember 2.12
Deprecation warning:

```
DEPRECATION: Using "_lookupFactory" is deprecated. Please use container.factoryFor instead. [deprecation id: container-lookupFactory] See http://emberjs.com/deprecations/v2.x/#toc_migrating-from-_lookupfactory-to-factoryfor for more details.
```

### PR DESCRIPTION
**Pull request by thoov**
Fixes #67

### CODE REVIEW

File: addon/state-for.js [hunk 1]
```diff
@@ -63,22 +63,24 @@ export default function stateFor(stateName, propertyName) {
  */
 function createStateFor(context, stateName, owner) {
```
Comment (thoov) [inline addon/state-for.js]: This file is the only one that is related to #67 changes

**Comment by jherdman** [comment]
@thoov I'm interesting in getting this pull request completed. May I be of assistance?

**Comment by thoov** [comment]
@jherdman I fixed the issue (I forgot to add back the getowner polyfil). I will release a new version shortly and can you verify in your app that everything looks good?

**Comment by thoov** [comment]
@jherdman I just published v3.4.0 which contains this fix

**Comment by mike183** [comment]
@thoov: Thanks for fixing this! I've tested v3.4.0 in my app and everything is working as expected.

**Pull request merged**